data/meson.build

```
@@ -0,0 +1,31 @@
+# icons
+icon_dir = join_paths(datadir, 'icons', 'hicolor')
+
+# scalable
+scalable = join_paths('icons', 'hicolor', 'scalable', 'apps', project_name + '.svg')
+install_data(scalable, install_dir: join_paths(icon_dir, 'scalable', 'apps'))
+
+# regular icons
+foreach size: ['16x16', '22x22', '24x24', '32x32', '48x48']
+  icon_name = join_paths('icons', 'hicolor', size, 'apps', project_name + '.png')
+  install_data(icon_name, install_dir: join_paths(icon_dir, size, 'apps'))
+endforeach
+
+
+po_dir = join_paths(meson.project_source_root(), 'po')
+# desktop file
+i18n.merge_file(input: '@0@.desktop.in'.format(app_id),
+                output: '@0@.desktop'.format(app_id),
+                install: true,
+                install_dir: join_paths(datadir, 'applications'),
+                po_dir: po_dir,
+                type: 'desktop')
+
+
+# appdata
+i18n.merge_file(input: '@0@.appdata.xml.in'.format(app_id),
+                output: '@0@.appdata.xml'.format(app_id),
+                install: true,
+                install_dir: join_paths(datadir, 'appdata'),
```

```suggestion
                install_dir: join_paths(datadir, 'metainfo'),
```

Both work, metainfo is just the newer recommendation.